backend/core/src/main/java/com/knewless/core/author/AuthorService.java

```diff
@@ -123,4 +134,16 @@ public AuthorPublicDto getAuthorPublicDto(UUID authorId, UserPrincipal userPrinc
                 courses, articles, printFollowButton);
     }
 
+    public List<FavouriteAuthorResponseDto> getFavouriteAuthorsByIds(List<UUID> uuids) {
+        List<Author> authors = authorRepository.findAll();
+        authors.removeIf(c->!uuids.contains(c.getId()));
+        List<FavouriteAuthorResponseDto> result = new ArrayList<>();
+        List<Course> courses = courseRepository.findAll();
+        List<Path> paths = pathRepository.findAll();
```

Comment:
It's very inefficient to load all data in memory and then filter them. Try to avoid method `findAll` and reduce the number of rows you fetch from DB.